File: bin2dec.js
// Bináris szám átalakítása decimális számmá

const x = "1101";

console.log("Bináris szám:", x);
console.log();

let d = 0;
let helyiertek = 1;

// Helyiérték kiszámítása a legnagyobb helyiértéktől indulva
for (let i = 1; i < x.length; i++) {
    helyiertek *= 2;
}

for (let i = 0; i < x.length; i++) {
    const sz = x[i] === "1" ? 1 : 0;
    console.log("   Számjegy   :", sz);
    console.log("   Helyi érték:", helyiertek);
    const r = sz * helyiertek;
    console.log("      Részösszeg:", r);
    d += r;
    helyiertek = Math.floor(helyiertek / 2); // Egész osztás
}

console.log();
console.log("Decimális szám:", d);
console.log("__________");

File: bin2dec.py
# Bináris szám átalakítása decimális számmá

x = "1101"

print("Bináris szám:", x)
print()

d = 0
helyiertek = 1

# Helyiérték kiszámítása a legnagyobb helyiértéktől indulva
for i in range(1, len(x)):
    helyiertek *= 2

for i in range(len(x)):
    if x[i] == "1":
        sz = 1
    else:
        sz = 0

    print("   Számjegy   :", sz)
    print("   Helyi érték:", helyiertek)
    r = sz * helyiertek
    print("      Részösszeg:", r)
    d += r
    helyiertek //= 2  # Egész osztás, hogy ne legyen lebegőpontos

print()
print("Decimális szám:", d)
print("__________")
